cmd/apiserver/server_test.go

```diff
@@ -0,0 +1,462 @@
+			}
+		}
+		req, err := http.NewRequest("DELETE", "/calculations/delete/", nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+		vars := make(map[string]string)
+		if !tc.errorExpected {
+			vars = map[string]string{
+				"id": tc.initialCalculation[0].Name,
+			}
+		} else {
+			vars = map[string]string{
+				"id": "calc-expect-error",
+			}
+		}
+		req = mux.SetURLVars(req, vars)
+
+		req.Header.Set("Content-Type", "application/json")
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(s.deleteCalculation)
+		handler.ServeHTTP(rr, req)
+
+		if status := rr.Code; status == http.StatusOK {
+			logrus.Info(rr.Body)
+		} else if status != http.StatusOK {
+			logrus.Info(rr.Body)
+		}
+	}
+}
+
+func TestGetCalculations(t *testing.T) {
+	testCases := []struct {
+		id             string
+		newCalculation []calculationsv1.Calculation
+		expected       []calculationsv1.Calculation
+	}{
+		{
+			id: "get no calculation",
+		},
+		{
+			id: "get one calculations",
+			newCalculation: []calculationsv1.Calculation{
+				{
+					TypeMeta:   metav1.TypeMeta{Kind: "Calculation", APIVersion: "vega.io/v1"},
+					ObjectMeta: metav1.ObjectMeta{Name: "calc-1"},
+					Phase:      calculationsv1.CreatedPhase,
+					Status:     calculationsv1.CalculationStatus{StartTime: metav1.Time{Time: time.Date(2000, 0, 0, 0, 0, 0, 0, time.UTC)}},
+					Spec:       calculationsv1.CalculationSpec{Teff: 10000.0, LogG: 4.0},
+				},
+			},
+		},
+		{
+			id: "get two calculations",
+			newCalculation: []calculationsv1.Calculation{
+				{
+					TypeMeta:   metav1.TypeMeta{Kind: "Calculation", APIVersion: "vega.io/v1"},
+					ObjectMeta: metav1.ObjectMeta{Name: "calc-2"},
+					Phase:      calculationsv1.CreatedPhase,
+					Status:     calculationsv1.CalculationStatus{StartTime: metav1.Time{Time: time.Date(2000, 0, 0, 0, 0, 0, 0, time.UTC)}},
+					Spec:       calculationsv1.CalculationSpec{Teff: 11000.0, LogG: 4.0},
+				},
+				{
+					TypeMeta:   metav1.TypeMeta{Kind: "Calculation", APIVersion: "vega.io/v1"},
+					ObjectMeta: metav1.ObjectMeta{Name: "calc-3"},
+					Phase:      calculationsv1.CreatedPhase,
+					Status:     calculationsv1.CalculationStatus{StartTime: metav1.Time{Time: time.Date(2000, 0, 0, 0, 0, 0, 0, time.UTC)}},
+					Spec:       calculationsv1.CalculationSpec{Teff: 12000.0, LogG: 4.0},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		fakecs := fake.NewSimpleClientset()
+		fakeClient := fakecs.VegaV1()
+
+		s := server{
+			logger: logrus.WithField("test-name", tc.id),
+			ctx:    context.Background(),
+			client: fakeClient,
+		}
+
+		for _, calc := range tc.newCalculation {
+			_, err := fakeClient.Calculations().Create(s.ctx, &calc, metav1.CreateOptions{})
+			if err != nil {
+				t.Errorf("Coudn't create the calculation: %v", tc.newCalculation)
+			} else {
+				logrus.WithFields(logrus.Fields{"teff": calc.Spec.Teff, "logG": calc.Spec.LogG}).Info("Calculation successfully created")
+			}
+			tc.expected = append(tc.expected, calc)
```

Comment:
Please add the `expected` list in the testcase struct. We need to know exactly what we should expect when someone sees the testcase directly. There is no need for this kind of mechanic when we write a test.